src/components/Checkbox/index.tsx

```diff
@@ -20,18 +22,31 @@ interface CheckboxUIProps extends CheckboxSharedProps {
 }
 
 export function CheckboxUI(props: CheckboxUIProps): ReactElement {
-    const { checked, className = '', kind = 'checkbox', label, name, onChange } = props;
-    const checkBoxStyles = checked ? 'bg-blue-500 shadow text-white' : 'border border-gray-300';
+    const { checked, className = '', kind = 'checkbox', label, name, onChange, required = false } = props;
+    const error = !checked && required;
+    const checkBoxStyles = checked
+        ? 'bg-blue-500 shadow text-white group-hover:bg-blue-600'
+        : `border ${
+              error ? 'border-red-600 group-hover:border-red-700' : 'border-gray-300 group-hover:border-gray-400'
+          }`;
     let radius = 'rounded-sm';
     if (kind === 'radio') {
         radius = 'rounded-xl';
     }
     return (
-        <div className={`cursor-pointer flex items-center ${className} checkbox`} data-tname={name} onClick={onChange}>
+        <div
+            className={`group cursor-pointer flex items-center ${className} checkbox`}
+            data-tname={name}
+            onClick={onChange}
+        >
             <div className={`flex flex-shrink-0 h-4 items-center justify-center ${radius} w-4 ${checkBoxStyles}`}>
```

Comment:
I missed this in previous PR apparently... same comment re: how to candle variable classNames

Reply:
the classNames util can take unlimited args, and can use an object for conditional classes too i.e.` classNames({ 'h-10': x === 10}, var1, var2, var3, 'w-10 justify-center')`